### PR TITLE
Add helm values schema

### PR DESCRIPTION
**Pull request by tfussell**
Towards https://github.com/giantswarm/giantswarm/issues/21417

I created the initial schema using `helm schema-gen values.yaml > values.schema.json` and then modified it to be more strict using domain knowledge of OpenStack.

### CODE REVIEW

**Comment by tfussell** [comment]
Tested successfully with Tekton.

File: helm/cluster-openstack/values.schema.json [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+        "clusterName",
+        "controlPlane",
+        "kubernetesVersion",
+        "managementCluster",
+        "networkName",
+        "nodeCIDR",
+        "nodeClasses",
+        "nodePools",
+        "organization",
+        "subnetName"
```
Comment (pipo02mix) [inline helm/cluster-openstack/values.schema.json:190]: `subnetName` is optional?

**Review by glitchcrab — APPROVED** [review]
I can't comment on the openstack-specific requirements, but looks good to me overall. No null values in sight :+1:

File: helm/cluster-openstack/values.schema.json [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+                "flavor": {
+                    "type": "string"
+                },
+                "image": {
+                    "type": "string"
+                }
+            },
+            "required": [
+                "flavor",
+                "image"
```
Comment (kopiczko) [inline helm/cluster-openstack/values.schema.json:27]: Isn't `diskSize` required?

File: helm/cluster-openstack/values.schema.json (hunk 1 above)
Comment (kopiczko) [reply helm/cluster-openstack/values.schema.json:190]: Smae for `networkName`.

File: helm/cluster-openstack/values.schema.json [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+                    "flavor": {
+                        "type": "string"
+                    },
+                    "image": {
+                        "type": "string"
+                    }
+                },
+                "required": [
+                    "flavor",
+                    "image"
```
Comment (kopiczko) [inline helm/cluster-openstack/values.schema.json:113]: Same here for `diskSize`.

File: helm/cluster-openstack/values.schema.json (hunk 1 above)
Comment (tfussell) [reply helm/cluster-openstack/values.schema.json:190]: Good point, I'll remove it.

File: helm/cluster-openstack/values.schema.json (hunk 2 above)
Comment (tfussell) [reply helm/cluster-openstack/values.schema.json:27]: AFAIK `diskSize` only applies if `bootFromVolume` is true (i.e. using root volumes). And we can't do dependent validation with JSON schema. @ericgraf Can you confirm that you can't set the disk size using ephemeral volumes?

Comment (kopiczko) [reply helm/cluster-openstack/values.schema.json:27]: Looks like you're right. Ephemeral root volume is defined by the flavour or image.

Comment (kopiczko) [reply helm/cluster-openstack/values.schema.json:27]: We could add some more sophisticated validation like here for that https://gigantic.slack.com/archives/C01BYMF6RN0/p1648218775082879

**Review by kopiczko — APPROVED** [review]

File: helm/cluster-openstack/values.schema.json (hunk 2 above)
Comment (tfussell) [reply helm/cluster-openstack/values.schema.json:27]: I feel like that kind of validation is separate from adding a values schema so I'll merge this PR without that.

**Pull request merged**